Index: src/main/java/com/ukmaSupport/models/PasswordPair.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/models/PasswordPair.java	(revision )
+++ src/main/java/com/ukmaSupport/models/PasswordPair.java	(revision )
@@ -0,0 +1,30 @@
+package com.ukmaSupport.models;
+
+
+
+public class PasswordPair {
+
+    private String password;
+
+    private String confPassword;
+
+
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public String getConfPassword() {
+        return confPassword;
+    }
+
+    public void setConfPassword(String confPassword) {
+        this.confPassword = confPassword;
+    }
+
+
+}
\ No newline at end of file
Index: src/main/java/com/ukmaSupport/utils/PasswordChangeValidator.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/utils/PasswordChangeValidator.java	(revision )
+++ src/main/java/com/ukmaSupport/utils/PasswordChangeValidator.java	(revision )
@@ -0,0 +1,17 @@
+package com.ukmaSupport.utils;
+
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+
+@Component
+public class PasswordChangeValidator {
+    public void validate(String pass,String confPass, Errors errors) {
+        ValidationUtils.rejectIfEmptyOrWhitespace(errors, "password", "valid.password", "Password is required.");
+        ValidationUtils.rejectIfEmptyOrWhitespace(errors, "confPassword", "valid.confPassword", "Confirm password is required.");
+
+        if (!pass.equals(confPass))
+            errors.rejectValue("confPassword", "valid.confPasswordDiff", "Passwords are different.");
+    }
+
+    }
Index: src/main/webapp/WEB-INF/views/userPage/passwordChangeSuccess.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/views/userPage/passwordChangeSuccess.jsp	(revision )
+++ src/main/webapp/WEB-INF/views/userPage/passwordChangeSuccess.jsp	(revision )
@@ -0,0 +1,24 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" pageEncoding="UTF-8" %>
+<%@ taglib prefix="form" uri="http://www.springframework.org/tags/form" %>
+<html>
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+    <title>Password change successful</title>
+</head>
+<body>
+<div align="center">
+    <table border="0">
+        <tr>
+            <td colspan="2" align="center"><h2>Password change successful</h2></td>
+        </tr>
+        <tr>
+            <td colspan="2" align="center">
+                <h3>You may now use the new pass to log in </h3>
+            </td>
+        </tr>
+
+
+    </table>
+</div>
+</body>
+</html>
\ No newline at end of file
Index: src/main/java/com/ukmaSupport/controllers/EditProfile.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/controllers/EditProfile.java	(revision )
+++ src/main/java/com/ukmaSupport/controllers/EditProfile.java	(revision )
@@ -0,0 +1,54 @@
+package com.ukmaSupport.controllers;
+
+import com.ukmaSupport.models.*;
+import com.ukmaSupport.services.interfaces.UserService;
+import com.ukmaSupport.utils.PasswordChangeValidator;
+import com.ukmaSupport.utils.PasswordEncryptor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+@Controller
+@RequestMapping(value = "/editProfile")
+public class EditProfile {
+
+    @Autowired
+    private UserService userDao;
+
+    @Autowired
+    @Qualifier("passChangeValidator")
+    private PasswordChangeValidator validator;
+
+    @RequestMapping(method = RequestMethod.GET)
+    public String viewRegistration(Model model) {
+       PasswordPair passwordPair = new PasswordPair();
+        model.addAttribute("passChangeForm", passwordPair);
+        return "userPage/editProfile";
+    }
+
+
+   @RequestMapping(method = RequestMethod.POST)
+    public String passChange(@ModelAttribute("passChangeForm") PasswordPair passwordPair, Model model, BindingResult result) {
+
+
+        validator.validate(passwordPair.getPassword(), passwordPair.getConfPassword(), result);
+        if (result.hasErrors())
+            return "userPage/editProfile";
+        String pass = PasswordEncryptor.encode(passwordPair.getPassword());
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        String name = auth.getName();
+        User user = userDao.getByEmail(name);
+        user.setPassword(pass);
+        userDao.saveOrUpdate(user);
+
+
+        return "userPage/passwordChangeSuccess";
+    }
+}
\ No newline at end of file
Index: src/main/webapp/WEB-INF/views/userPage/editProfile.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/views/userPage/editProfile.jsp	(date 1447366407000)
+++ src/main/webapp/WEB-INF/views/userPage/editProfile.jsp	(revision )
@@ -1,9 +1,95 @@
-<%@ page contentType="text/html;charset=UTF-8" language="java" %>
-<html>
+<%@ page language="java" pageEncoding="UTF-8" contentType="text/html;charset=utf-8" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ taglib prefix="form" uri="http://www.springframework.org/tags/form" %>
+<%@ taglib prefix="spring" uri="http://www.springframework.org/tags" %>
+<html xmlns:th="http://www.thymeleaf.org">
 <head>
-    <title>Eit Profile</title>
+    <meta http-equiv = "Content-Type" content="text/html; charset=utf-8" />
+    <title>Profile edit</title>
+    <link rel="Shortcut Icon" href="" type="image/x-icon" />
+    <link rel="stylesheet" href="../../../resources/css/bootstrap.css">
+    <link rel="stylesheet" href="../../../resources/css/main.css" type="text/css" media="screen" />
+    <script src="../../../resources/js/jquery-1.11.3.js"></script>
+    <script src="../../../resources/js/bootstrap.min.js"></script>
+    <script>
+        jQuery( function($) {
+            $('tbody tr[data-href]').addClass('clickable').click( function() {
+                window.location = $(this).attr('data-href');
+            });
+        });
+    </script>
 </head>
+
 <body>
 
+<div id="wrap">
+    <nav id="header">
+        <div class="container-fluid">
+            <div class="navbar-header">
+                <a href=""><img id="logo" alt="brand" src="../../../resources/img/logo.png"></a>
+            </div>
+            <div class="collapse navbar-collapse">
+                <ul class="nav navbar-nav navbar-right">
+                    <li class="dropdown">
+                        <a class="dropdown-toggle menu-element" data-toggle="dropdown" href="#"> My orders<b class="caret"></b></a>
+                        <ul class="dropdown-menu">
+                            <li class="drop-menu-element"><a class="menu-element-li" href="/complited">Completed orders</a></li>
+                            <li class="drop-menu-element"><a class="menu-element-li" href="/uncomplited">Uncompleted orders</a></li>
+                        </ul>
+                    </li>
+                    <li><a id = "editProfile" class="menu-element" href="/editProfile">Edit profile</a></li>
+                    <li><a class="menu-element" href="#">Log out</a></li>
+                </ul>
+            </div>
+        </div>
+    </nav>
+
+    <div>
+        <p id="hello">Hello, you may edit your e-mail or your password here.</p>
+    </div>
+
+    <div id="profEdit">
+        <form:form id="passChangeForm" action="editProfile" method="post" commandName="passChangeForm">
+            <table border="0">
+                <tr>
+                    <td class="centralWord" colspan="2" align="center">Password change</td>
+                </tr>
+
+                <tr>
+                    <td><label class="labels">New password:</label></td>
+                    <td><form:input path = "password" type="password" value=""
+                                    class="form-control form-style"/></td>
+                    <td><form:errors path = "password" class="regErrors" id="password.errors"
+                                     cssStyle="color: #ff0000;"/></td>
+                </tr>
+                <tr>
+                    <td><label class="labels">Confirm new password:</label></td>
+                    <td><form:input path = "confPassword" type="password" value=""
+                                    class="form-control form-style"/></td>
+                    <td><form:errors path = "confPassword" class="regErrors" id="confPassword.errors"
+                                     cssStyle="color: #ff0000;"/></td>
+
+                </tr>
+
+            </table>
+            <div class="form-group">
+                <div class="col-md-offset-1  col-md-4">
+                    <button id="change-pass-button" type="submit" value="editProfile"
+                            class="btn btn-primary btn-block button-style">Change Password
+                    </button>
+                </div>
+            </div>
+        </form:form>
+    </div>
+
+
+
+    <div class="navbar-fixed-bottom">
+        <div class="thick"></div>
+        <div class="thin"></div>
+        <div><p class="footertext"> © 2015 All Rights Reserved</p></div>
+    </div>
+
+</div>
 </body>
 </html>
Index: src/main/java/com/ukmaSupport/controllers/LabAssistController.java
===================================================================
--- src/main/java/com/ukmaSupport/controllers/LabAssistController.java	(date 1447366407000)
+++ src/main/java/com/ukmaSupport/controllers/LabAssistController.java	(date 1447366407000)
@@ -1,78 +0,0 @@
-package com.ukmaSupport.controllers;
-
-import com.ukmaSupport.models.Order;
-import com.ukmaSupport.services.interfaces.OrderService;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Controller;
-import org.springframework.ui.Model;
-import org.springframework.ui.ModelMap;
-import org.springframework.validation.BindingResult;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.context.request.RequestContextHolder;
-import org.springframework.web.context.request.ServletRequestAttributes;
-
-import javax.servlet.http.HttpSession;
-import java.util.List;
-
-
-@Controller
-public class LabAssistController {
-
-    @Autowired
-    private OrderService orderDao;
-
-    @Autowired
-    private UserController userController;
-
-
-    @RequestMapping(value = "/allAssistOrders", method = RequestMethod.GET)
-    public String showAllAssistOrders(Model model) {
-        ServletRequestAttributes attr = (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes();
-        HttpSession session = attr.getRequest().getSession();
-        int assistid = (Integer) session.getAttribute("id");
-
-        List<Order> orders = orderDao.getByAssist(assistid);
-        model.addAttribute("assistOrders", orders);
-        return "assistPage/assistHomePage";
-    }
-
-    @RequestMapping(value = "/assistCompletedOrders", method = RequestMethod.GET)
-    public String showAsistCompletedOrders(Model model) {
-        ServletRequestAttributes attr = (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes();
-        HttpSession session = attr.getRequest().getSession();
-        int assistid = (Integer) session.getAttribute("id");
-
-        List<Order> orders = orderDao.getByAssistAndStatus(assistid, "Done");
-        model.addAttribute("asistCompletedOrders", orders);
-        return "assistPage/assistHomePage";
-    }
-
-    @RequestMapping(value = "/assistUncompletedOrders", method = RequestMethod.GET)
-    public String showAssistUncompletedOrders(Model model) {
-        ServletRequestAttributes attr = (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes();
-        HttpSession session = attr.getRequest().getSession();
-        int assistid = (Integer) session.getAttribute("id");
-
-        List<Order> orders = orderDao.getByAssistAndStatus(assistid, "Undone");
-        model.addAttribute("asistUncompletedOrders", orders);
-        return "assistPage/assistHomePage";
-    }
-
-    @RequestMapping(value = "/editAssistProfile" , method = RequestMethod.GET)
-    public String editProfile(ModelMap model) {
-       return userController.editProfile(model);
-    }
-
-    @RequestMapping(value = "/createAssistOrder" , method = RequestMethod.GET)
-    public String createAssistOrder(ModelMap model) {
-        return userController.createOrder(model);
-    }
-
-    @RequestMapping(value = "/createAssistOrder" , method = RequestMethod.POST)
-    public String createAssistOrderPost(Order order,ModelMap model, BindingResult result) {
-        return userController.createOrderPost(order, model, result);
-    }
-
-
-}
Index: src/main/java/com/ukmaSupport/controllers/Registration.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/controllers/Registration.java	(date 1447366407000)
+++ src/main/java/com/ukmaSupport/controllers/Registration.java	(revision )
@@ -1,11 +1,11 @@
 package com.ukmaSupport.controllers;
 
 import com.ukmaSupport.mailService.templates.RegistrationMail;
-import com.ukmaSupport.models.User;
+import com.ukmaSupport.models.*;
 import com.ukmaSupport.services.interfaces.UserService;
 import com.ukmaSupport.utils.Constants;
-import com.ukmaSupport.utils.PasswordEncryptor;
 import com.ukmaSupport.utils.RegistrationValidator;
+import com.ukmaSupport.utils.PasswordEncryptor;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Controller;
\ No newline at end of file
Index: src/main/webapp/WEB-INF/views/adminPage/users.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/views/adminPage/users.jsp	(date 1447366407000)
+++ src/main/webapp/WEB-INF/views/adminPage/users.jsp	(revision )
@@ -28,6 +28,14 @@
                         </ul>
                     </li>
                     <li class="dropdown">
+                        <a class="dropdown-toggle menu-element" data-toggle="dropdown" href="#">A<b
+                                class="caret"></b></a>
+                        <ul class="dropdown-menu">
+                            <li><a class="menu-element-li" href="/createAuditorium">Add Auditorium</a></li>
+                            <li><a class="menu-element-li" href="/auditoriums">Auditoriums</a></li>
+                        </ul>
+                    </li>
+                    <li class="dropdown">
                         <a class="dropdown-toggle menu-element active" data-toggle="dropdown" href="#">Users<b
                                 class="caret"></b></a>
                         <ul class="dropdown-menu">
@@ -47,7 +55,7 @@
                             <li><a class="menu-element-li" href="/uncompletedOrders">Uncomplited orders</a></li>
                         </ul>
                     </li>
-                    <li><a class="menu-element" href="#">Edit profile</a></li>
+                    <li><a id = "editProfile" class="menu-element" href="/editProfile">Edit profile</a></li>
                     <li><a class="menu-element" href="#">Log out</a></li>
                 </ul>
             </div>
Index: src/main/webapp/WEB-INF/views/assistPage/assistHomePage.jsp
===================================================================
--- src/main/webapp/WEB-INF/views/assistPage/assistHomePage.jsp	(date 1447366407000)
+++ src/main/webapp/WEB-INF/views/assistPage/assistHomePage.jsp	(date 1447366407000)
@@ -1,9 +0,0 @@
-<%@ page contentType="text/html;charset=UTF-8" language="java" %>
-<html>
-<head>
-    <title>Home</title>
-</head>
-<body>
-
-</body>
-</html>
Index: src/main/java/com/ukmaSupport/controllers/AdminController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/controllers/AdminController.java	(date 1447366407000)
+++ src/main/java/com/ukmaSupport/controllers/AdminController.java	(revision )
@@ -98,6 +98,29 @@
 
     @RequestMapping(value = "/auditoriums", method = RequestMethod.GET)
     public String showAuditoriums(Model model) {
+        List<Auditorium> auditoriums = auditoriumDao.getAll();
+        model.addAttribute("auditoriums", auditoriums);
+        return "adminPage/auditoriums";
+    }
+
+    @RequestMapping(value = "/createAuditorium", method = RequestMethod.GET)
+    public String createAuditorium(ModelMap model) {
+        Auditorium order = new Auditorium();
+        List<User> users = userDao.getByRole("ASSISTANT");
+        model.addAttribute("newAuditorium", order);
+        model.addAttribute("assistants", users);
+        return "adminPage/addAuditorium";
+    }
+
+    @RequestMapping(value = "/createAuditorium", method = RequestMethod.POST)
+    public String saveAuditorium(@ModelAttribute("newAuditorium") Auditorium auditorium, ModelMap model) {
+        model.addAttribute("number", auditorium.getNumber());
+        auditoriumDao.save(auditorium);
+        return "redirect:/all";
+    }
+
+    @RequestMapping(value = "/auditoriums", method = RequestMethod.GET)
+    public String showAuditoriums(Model model) {
         List<Auditorium> auditoriums = auditoriumService.getAll();
         model.addAttribute("auditoriums", auditoriums);
         return "adminPage/auditoriums";
\ No newline at end of file
Index: src/main/java/com/ukmaSupport/services/SpringSecurityEncoder.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/services/SpringSecurityEncoder.java	(date 1447366407000)
+++ src/main/java/com/ukmaSupport/services/SpringSecurityEncoder.java	(revision )
@@ -1,15 +1,14 @@
 package com.ukmaSupport.services;
 
-import com.ukmaSupport.models.User;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
-
-import javax.crypto.SecretKeyFactory;
-import javax.crypto.spec.PBEKeySpec;
 import java.math.BigInteger;
 import java.security.NoSuchAlgorithmException;
 import java.security.SecureRandom;
 import java.security.spec.InvalidKeySpecException;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import com.ukmaSupport.models.User;
 
 
 
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- pom.xml	(date 1447366407000)
+++ pom.xml	(revision )
@@ -17,6 +17,7 @@
 
     <dependencies>
 
+
         <!-- Spring Security -->
         <dependency>
             <groupId>org.springframework.security</groupId>
@@ -138,6 +139,7 @@
             <artifactId>jackson-databind</artifactId>
             <version>${jackson.version}</version>
         </dependency>
+
 
 
     </dependencies>
\ No newline at end of file
Index: src/main/webapp/WEB-INF/views/adminPage/auditoriumsList.jsp
===================================================================
--- src/main/webapp/WEB-INF/views/adminPage/auditoriumsList.jsp	(date 1447366407000)
+++ src/main/webapp/WEB-INF/views/adminPage/auditoriumsList.jsp	(date 1447366407000)
@@ -1,86 +0,0 @@
-<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
-<%@ page contentType="text/html;charset=UTF-8" language="java" pageEncoding="UTF-8" %>
-<html lang="en">
-<head>
-    <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
-    <title>Admin Page List Users</title>
-    <link rel="Shortcut Icon" href="" type="image/x-icon"/>
-    <link rel="stylesheet" href="../../../resources/css/bootstrap.css">
-    <link rel="stylesheet" href="../../../resources/css/main.css" type="text/css" media="screen"/>
-</head>
-
-<body>
-
-<div id="wrap">
-    <nav id="header">
-        <div class="container-fluid">
-            <div class="navbar-header">
-                <a href=""><img id="logo" alt="brand" src="../../../resources/img/logo.png"></a>
-            </div>
-            <div class="collapse navbar-collapse">
-                <ul class="nav navbar-nav navbar-right">
-                    <li class="dropdown">
-                        <a class="dropdown-toggle menu-element active" data-toggle="dropdown" href="#">A<b
-                                class="caret"></b></a>
-                        <ul class="dropdown-menu">
-                            <li><a class="menu-element-li" href="/addAuditorium">Add Auditorium</a></li>
-                            <li><a class="menu-element-li" href="/auditoriums">Auditoriums</a></li>
-                        </ul>
-                    </li>
-                    <li class="dropdown">
-                        <a class="dropdown-toggle menu-element" data-toggle="dropdown" href="#">Users<b
-                                class="caret"></b></a>
-                        <ul class="dropdown-menu">
-                            <li><a class="menu-element-li" href="/all">All</a></li>
-                            <li><a class="menu-element-li" href="/users">Users</a></li>
-                            <li><a class="menu-element-li" href="/assistants">Assistants</a></li>
-                            <li><a class="menu-element-li" href="/professors">Professors</a></li>
-                            <li><a class="menu-element-li" href="/blocked">Blocked users</a></li>
-                        </ul>
-                    </li>
-                    <li class="dropdown">
-                        <a class="dropdown-toggle menu-element" data-toggle="dropdown" href="#">Orders<b
-                                class="caret"></b></a>
-                        <ul class="dropdown-menu">
-                            <li><a class="menu-element-li" href="/allOrders">All orders</a></li>
-                            <li><a class="menu-element-li" href="/completedOrders">Complited orders</a></li>
-                            <li><a class="menu-element-li" href="/uncompletedOrders">Uncomplited orders</a></li>
-                        </ul>
-                    </li>
-                    <li><a class="menu-element" href="#">Edit profile</a></li>
-                    <li><a class="menu-element" href="#">Log out</a></li>
-                </ul>
-            </div>
-        </div>
-    </nav>
-    <div>
-        <p id="hello">Hello, Admin!</p>
-    </div>
-
-    <div class="table-align">
-        <table class="tbl table table-striped">
-            <thead>
-            <tr>
-                <th>Name</th>
-                <%--<th>Assistant</th>--%>
-            </tr>
-            </thead>
-            <tbody>
-            <c:forEach items="${auditoriums}" var="item" varStatus="count">
-                <tr data-href="#">
-                    <td>${item.number}</td>
-                </tr>
-            </c:forEach>
-            </tbody>
-        </table>
-    </div>
-</div>
-<div class="navbar-fixed-bottom">
-    <div class="thick"></div>
-    <div class="thin"></div>
-    <div><p class="footertext"> © 2015 All Rights Reserved</p></div>
-</div>
-<script src="../../../resources/js/jquery-1.11.3.js"></script>
-<script src="../../../resources/js/bootstrap.min.js"></script>
-</body>
-</html>
\ No newline at end of file
Index: src/main/webapp/WEB-INF/views/userPage/userHomePage.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/views/userPage/userHomePage.jsp	(date 1447366407000)
+++ src/main/webapp/WEB-INF/views/userPage/userHomePage.jsp	(revision )
@@ -33,11 +33,11 @@
                     <li class="dropdown">
                         <a class="dropdown-toggle menu-element" data-toggle="dropdown" href="#"> My orders<b class="caret"></b></a>
                         <ul class="dropdown-menu">
-                            <li class="drop-menu-element"><a class="menu-element-li" href="/completed">Completed orders</a></li>
-                            <li class="drop-menu-element"><a class="menu-element-li" href="/uncompleted">Uncompleted orders</a></li>
+                            <li class="drop-menu-element"><a class="menu-element-li" href="/complited">Complited orders</a></li>
+                            <li class="drop-menu-element"><a class="menu-element-li" href="/uncomplited">Uncomplited orders</a></li>
                         </ul>
                     </li>
-                    <li><a class="menu-element" href="#">Edit profile</a></li>
+                    <li><a id = "editProfile" class="menu-element" href="/editProfile">Edit profile</a></li>
                     <li><a class="menu-element" href="#">Log out</a></li>
                 </ul>
             </div>
Index: src/main/webapp/WEB-INF/views/adminPage/auditoriums.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/views/adminPage/auditoriums.jsp	(date 1447366407000)
+++ src/main/webapp/WEB-INF/views/adminPage/auditoriums.jsp	(revision )
@@ -47,7 +47,7 @@
                             <li><a class="menu-element-li" href="/uncompletedOrders">Uncomplited orders</a></li>
                         </ul>
                     </li>
-                    <li><a class="menu-element" href="#">Edit profile</a></li>
+                    <li><a id = "editProfile" class="menu-element" href="/editProfile">Edit profile</a></li>
                     <li><a class="menu-element" href="#">Log out</a></li>
                 </ul>
             </div>
Index: src/main/webapp/WEB-INF/spring/security.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/spring/security.xml	(date 1447366407000)
+++ src/main/webapp/WEB-INF/spring/security.xml	(revision )
@@ -51,6 +51,7 @@
 
         <intercept-url pattern="/" access="permitAll"/>
         <intercept-url pattern="/userhome**" access="hasRole('ROLE_USER')"/>
+        <intercept-url pattern="/editProfile**" access="hasRole('ROLE_USER')"/>
         <intercept-url pattern="/admin**" access="hasRole('ROLE_ADMIN')"/>
 
     </http>
\ No newline at end of file
Index: src/main/java/com/ukmaSupport/controllers/UserController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/controllers/UserController.java	(date 1447366407000)
+++ src/main/java/com/ukmaSupport/controllers/UserController.java	(revision )
@@ -35,6 +35,7 @@
     private WorkplaceService workplaceService;
 
 
+
     @RequestMapping(value = "/ajaxtest", method = RequestMethod.GET)
     public @ResponseBody List<Workplace> getCharNum(@RequestParam("text") String text) {
         System.out.println(text);
@@ -65,7 +66,7 @@
         return "redirect:/userhome";
     }
    @RequestMapping(value = "/userhome" , method = RequestMethod.GET)
-    public String listUsersOrders(ModelMap model) {
+    public String listUsersOrder(ModelMap model) {
         ServletRequestAttributes attr = (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes();
         HttpSession session = attr.getRequest().getSession();
         int userid = (Integer) session.getAttribute("id");
@@ -73,13 +74,11 @@
         System.out.println(userid);
         List<Order> orders =  orderService.getByUserId(userid);
         model.addAttribute("userOrder", orders);
-        model.addAttribute("message", "Gt");
-
         return "userPage/userHomePage";
     }
 
-    @RequestMapping(value = "/uncompleted" , method = RequestMethod.GET)
-    public String uncompletedOrders(ModelMap model) {
+    @RequestMapping(value = "/uncomplited" , method = RequestMethod.GET)
+    public String uncomlitedOrder(ModelMap model) {
         ServletRequestAttributes attr = (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes();
         HttpSession session = attr.getRequest().getSession();
         int userid = (Integer) session.getAttribute("id");
@@ -90,8 +89,8 @@
         return "userPage/userHomePage";
     }
 
-    @RequestMapping(value = "/completed" , method = RequestMethod.GET)
-    public String completedOrders(ModelMap model) {
+    @RequestMapping(value = "/complited" , method = RequestMethod.GET)
+    public String comlitedOrder(ModelMap model) {
         ServletRequestAttributes attr = (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes();
         HttpSession session = attr.getRequest().getSession();
 
Index: src/main/webapp/WEB-INF/views/registration/registration.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/views/registration/registration.jsp	(date 1447366407000)
+++ src/main/webapp/WEB-INF/views/registration/registration.jsp	(revision )
@@ -70,7 +70,7 @@
             </table>
             <div class="form-group">
                 <div class="col-md-offset-7 col-md-4">
-                    <button id="register-button" type="submit" value="Register"
+                    <button id="register-button" type="submit" value="register"
                             class="btn btn-primary btn-block button-style">Register
                     </button>
                 </div>
Index: src/main/java/com/ukmaSupport/dao/impl/AuditoriumDaoImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/dao/impl/AuditoriumDaoImpl.java	(date 1447366407000)
+++ src/main/java/com/ukmaSupport/dao/impl/AuditoriumDaoImpl.java	(revision )
@@ -1,8 +1,8 @@
 package com.ukmaSupport.dao.impl;
 
-import com.ukmaSupport.dao.interfaces.AuditoriumDao;
-import com.ukmaSupport.dao.mapper.AuditoriumMapper;
 import com.ukmaSupport.models.Auditorium;
+import com.ukmaSupport.dao.mapper.AuditoriumMapper;
+import com.ukmaSupport.dao.interfaces.AuditoriumDao;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Repository;
Index: src/main/java/com/ukmaSupport/services/interfaces/UserService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/services/interfaces/UserService.java	(date 1447366407000)
+++ src/main/java/com/ukmaSupport/services/interfaces/UserService.java	(revision )
@@ -6,8 +6,6 @@
 
 public interface UserService {
 
-    /**/
-
     User getById(int id);
 
     void delete(int id);
Index: src/main/java/com/ukmaSupport/services/UserServices.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/services/UserServices.java	(date 1447366407000)
+++ src/main/java/com/ukmaSupport/services/UserServices.java	(revision )
@@ -1,9 +1,9 @@
 package com.ukmaSupport.services;
 
 import com.ukmaSupport.services.interfaces.UserService;
+import org.springframework.security.core.userdetails.User;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.core.userdetails.User;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
Index: src/main/java/com/ukmaSupport/dao/impl/OrderDaoImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/ukmaSupport/dao/impl/OrderDaoImpl.java	(date 1447366407000)
+++ src/main/java/com/ukmaSupport/dao/impl/OrderDaoImpl.java	(revision )
@@ -8,10 +8,7 @@
 import org.springframework.jdbc.core.RowMapper;
 import org.springframework.stereotype.Repository;
 
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
+import java.sql.*;
 import java.util.List;
 
 @Repository("orderDao")
Index: src/main/webapp/WEB-INF/mvc-dispatcher-servlet.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/mvc-dispatcher-servlet.xml	(date 1447366407000)
+++ src/main/webapp/WEB-INF/mvc-dispatcher-servlet.xml	(revision )
@@ -24,7 +24,7 @@
     <context:annotation-config/>
 
 
-    <!-- error messages and validator for registration -->
+    <!-- error messages and validators for registration and password change -->
     <bean id="messageSource"
           class="org.springframework.context.support.ResourceBundleMessageSource">
         <property name="basename" value="messages"/>
@@ -32,6 +32,7 @@
     </bean>
 
     <bean id="registrationValidator" class="com.ukmaSupport.utils.RegistrationValidator"/>
+    <bean id="passChangeValidator" class="com.ukmaSupport.utils.PasswordChangeValidator"/>
     <!--><-->
 
     <!-- jdbcTemplate uses dataSource -->
\ No newline at end of file
Index: src/main/webapp/WEB-INF/views/adminPage/orders.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/views/adminPage/orders.jsp	(date 1447366407000)
+++ src/main/webapp/WEB-INF/views/adminPage/orders.jsp	(revision )
@@ -28,6 +28,14 @@
                         </ul>
                     </li>
                     <li class="dropdown">
+                        <a class="dropdown-toggle menu-element" data-toggle="dropdown" href="#">A<b
+                                class="caret"></b></a>
+                        <ul class="dropdown-menu">
+                            <li><a class="menu-element-li" href="/createAuditorium">Add Auditorium</a></li>
+                            <li><a class="menu-element-li" href="/auditoriums">Auditoriums</a></li>
+                        </ul>
+                    </li>
+                    <li class="dropdown">
                         <a class="dropdown-toggle menu-element" data-toggle="dropdown" href="#">Users<b
                                 class="caret"></b></a>
                         <ul class="dropdown-menu">
@@ -47,7 +55,7 @@
                             <li><a class="menu-element-li" href="/uncompletedOrders">Uncomplited orders</a></li>
                         </ul>
                     </li>
-                    <li><a class="menu-element" href="#">Edit profile</a></li>
+                    <li><a id = "editProfile" class="menu-element" href="/editProfile">Edit profile</a></li>
                     <li><a class="menu-element" href="#">Log out</a></li>
                 </ul>
             </div>
